pkg/ecdsa/signature.go

```diff
@@ -18,10 +18,17 @@ func EmptySignature(group curve.Curve) Signature {
 func (sig Signature) Verify(X curve.Point, hash []byte) bool {
 	group := X.Curve()
 
+	r := sig.R.XScalar()
+
+	if r.IsZero() || sig.S.IsZero() {
+		return false
+	}
+
+	// TODO Do we also need to check for R or S > the group modulus?
```

Comment:
For S, the unmarshalling process already checks that it's less than the modulus: https://github.com/decred/dcrd/blob/master/dcrec/secp256k1/modnscalar.go#L320, https://github.com/taurusgroup/multi-party-sig/blob/1c20cbbca313a5bbbdbeae497144294658257100/pkg/math/curve/secp256k1.go#L97.